src/main/java/ch/jalu/configme/beanmapper/propertydescription/BeanDescriptionFactoryImpl.java

```diff
@@ -3,89 +3,113 @@
 import ch.jalu.configme.Comment;
 import ch.jalu.configme.beanmapper.ConfigMeMapperException;
 import ch.jalu.configme.beanmapper.ExportName;
-import ch.jalu.typeresolver.TypeInfo;
+import ch.jalu.configme.beanmapper.IgnoreInMapping;
+import ch.jalu.configme.exception.ConfigMeException;
+import ch.jalu.configme.internal.record.RecordComponent;
+import ch.jalu.typeresolver.reflect.FieldUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-import java.beans.IntrospectionException;
-import java.beans.Introspector;
-import java.beans.PropertyDescriptor;
 import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.HashMap;
 import java.util.HashSet;
+import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Objects;
 import java.util.Set;
 import java.util.UUID;
-import java.util.stream.Collectors;
 
 /**
  * Creates all {@link BeanPropertyDescription} objects for a given class.
  * <p>
- * This description factory returns property descriptions for all properties on a class
- * for which a getter and setter is associated. Inherited properties are considered.
+ * This description factory returns property descriptions for all instance fields on a class,
+ * including fields on its parent. If a class has a field of the same name as the parent, the parent's
+ * field is ignored.
```

Comment:
maybe turn the meaning around, that a child field overrides a parent field if it has the same name